app/src/main/java/com/example/todolistproject/view/TaskActivity.kt

```diff
@@ -0,0 +1,243 @@
+package com.example.todolistproject.view
+
+import android.app.DatePickerDialog
+import android.content.DialogInterface
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.text.InputType
+import android.view.LayoutInflater
+import android.view.View
+import androidx.appcompat.app.AlertDialog
+import androidx.databinding.DataBindingUtil
+import androidx.activity.viewModels
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.todolistproject.R
+import com.example.todolistproject.databinding.ActivityMainBinding
+import com.example.todolistproject.databinding.DialogAddTaskBinding
+import com.example.todolistproject.model.Task
+import com.example.todolistproject.view.adapter.TaskAdapter
+import com.example.todolistproject.viewmodel.TaskViewModel
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import com.google.android.material.textfield.TextInputEditText
+import dagger.hilt.android.AndroidEntryPoint
+import java.text.SimpleDateFormat
+import java.util.*
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class TaskActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityMainBinding
+
+    @Inject lateinit var mAdapter: TaskAdapter
+    private val mViewModel by viewModels<TaskViewModel>()
+    private var currentFilter = TASK
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
+        binding.activity = this@TaskActivity
+
+        initRecyclerView()
+        observeViewModel()
+    }
+
+    private fun initRecyclerView() {
+        mAdapter?.apply {
+            listener = object : TaskAdapter.OnTaskItemClickListener {
+                override fun onTaskItemClick(position: Int) {
+                    openModifyTaskDialog(getItem(position))
+                }
+
+                override fun onTaskItemLongClick(position: Int) {
+                    openDeleteTaskDialog(getItem(position))
+                }
+
+                override fun onTaskCompleteClick(position: Int) {
+                    setTaskCompleted(getItem(position))
+                }
+            }
+        }
+
+        binding.rvTodoList.run {
+            layoutManager = LinearLayoutManager(this@TaskActivity)
+            adapter = mAdapter
+        }
```

Comment:
```suggestion
        binding.rvTodoList.apply{
            layoutManager = LinearLayoutManager(this@TaskActivity)
            adapter = mAdapter
        }
```
취향에 문제일수도 있긴 한데 뉘앙스 적으로는 apply가 더어울릴듣 합니다.
list 로 run 한다 보다는 list 에 apply 한다 가좀더 부드러워 보여요

Reply:
LinearLayoutManager와 같은경우는 Binding에 `app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager` 를 활용할 수 있어요.